File: tests/interactive-shells/matlab-octave/01-DarkVaderTests/Tests20180119Dataset01/Tests011MeasuresCentralTendencies.m


if ~(exist("data", "var") == 1)
    # https://octave.org/doc/v4.4.1/Including-File-Content.html
    # <include>Tests00Data.m</include>
    # addpath("./interactive-shells/matlab-octave/01-DarkVaderTests/Tests20180119Dataset01/");
    # source ("Tests00Data.m", "caller")
    # run("./Tests00Data.m")
    # Tests00Data
    file_path = fileparts(mfilename('fullpath'));
    file_path
    addpath(file_path);
    run("Tests00Data.m");
endif


disp ("==================================================================================")
# Arrange
tic ();
# Act
mean_arithmetic = mean_arithmetic = mean(data, "a");
elapsed_time = toc ();
# Assert
disp ("mean_arithmetic               = "), disp(mean_arithmetic)
disp ("              time [\u00B5s] = "), disp(elapsed_time * 1000000)
#assert(mean_arithmetic == 3.8)
disp ("==================================================================================")


disp ("==================================================================================")
# Arrange
tic ();
# Act
mean_geometric = mean(data, "g");
elapsed_time = toc ();
# Assert
disp ("mean_geometric               = "), disp(mean_geometric)
disp ("              time [\u00B5s] = "), disp(elapsed_time * 1000000)
#assert(mean_arithmetic == 3.8)
disp ("==================================================================================")


disp ("==================================================================================")
# Arrange
tic ();
# Act
mean_harmonic = mean(data, "h");
elapsed_time = toc ();
# Assert
disp ("mean_harmonic                = "), disp(mean_harmonic)
disp ("              time [\u00B5s] = "), disp(elapsed_time * 1000000)
#assert(mean_arithmetic == 3.8)
disp ("==================================================================================")


disp ("==================================================================================")
# Arrange
tic ();
# Act
mean_squared = meansq(data);
elapsed_time = toc ();
# Assert
disp ("mean_squared                 = "), disp(mean_squared)
disp ("              time [\u00B5s] = "), disp(elapsed_time * 1000000)
#assert(mean_arithmetic == 3.8)
disp ("==================================================================================")

disp ("==================================================================================")
# Arrange
tic ();
# Act
result_median = median(data);
elapsed_time = toc ();
# Assert
disp ("median                       = "), disp(result_median)
disp ("              time [\u00B5s] = "), disp(elapsed_time * 1000000)
#assert(mean_arithmetic == 3.8)
disp ("==================================================================================")


disp ("==================================================================================")
# Arrange
tic ();
# Act
[M, F, C] = mode(data);
elapsed_time = toc ();
# Assert
disp ("mode                         = "), disp(M)
disp ("F                            = "), disp(F)
disp ("C                            = "), disp(C)
disp ("              time [\u00B5s] = "), disp(elapsed_time * 1000000)
#assert(mean_arithmetic == 3.8)
disp ("==================================================================================")


disp ("==================================================================================")
# Arrange
tic ();
# Act
result_range = range(data);
elapsed_time = toc ();
# Assert
disp ("range                        = "), disp(result_range)
disp ("              time [\u00B5s] = "), disp(elapsed_time * 1000000)
#assert(mean_arithmetic == 3.8)
disp ("==================================================================================")

File: tests/interactive-shells/python/01-DarkVaderTests/Tests20180119Dataset03/Tests011MeasuresCentralTendencies.py
from Tests00Data import *
from scipy import stats
import statistics
import timeit

# https://docs.python.org/3/library/statistics.html
# https://docs.scipy.org/doc/scipy/reference/stats.html


print ("==================================================================================")
# Arrange
start = timeit.default_timer()
# Act
mean_arithmetic = statistics.mean(data)
end = timeit.default_timer()
# Assert
print ("mean_arithmetic               = ", mean_arithmetic)
print (u"              time [\u00B5s] = ", (end - start) * 100000)
#assert(mean_arithmetic == 3.8)
print ("==================================================================================")

print ("==================================================================================")
# Arrange
start = timeit.default_timer()
# Act
mean_geometric = stats.gmean(data)
end = timeit.default_timer()
# Assert
print ("mean_geometric                = ", mean_geometric)
print (u"              time [\u00B5s] = ", (end - start) * 100000)
#assert(mean_arithmetic == 3.8)
print ("==================================================================================")

print ("==================================================================================")
# Arrange
start = timeit.default_timer()
# Act
mean_harmonic = stats.hmean(data)
end = timeit.default_timer()
# Assert
print ("mean_harmonic                 = ", mean_harmonic)
print (u"              time [\u00B5s] = ", (end - start) * 100000)
#assert(mean_arithmetic == 3.8)
print ("==================================================================================")

print ("==================================================================================")
# Arrange
start = timeit.default_timer()
# Act
median = statistics.median(data)
end = timeit.default_timer()
# Assert
print ("median                        = ", median)
print (u"              time [\u00B5s] = ", (end - start) * 100000)
#assert(mean_arithmetic == 3.8)
print ("==================================================================================")

print ("==================================================================================")
# Arrange
start = timeit.default_timer()
# Act
moment = stats.moment(data)
end = timeit.default_timer()
# Assert
print ("moment                        = ", moment)
print (u"              time [\u00B5s] = ", (end - start) * 100000)
#assert(mean_arithmetic == 3.8)
print ("==================================================================================")

print ("==================================================================================")
# Arrange
start = timeit.default_timer()
# Act
median_low = statistics.median_low(data)
end = timeit.default_timer()
# Assert
print ("median_low                    = ", median_low)
print (u"              time [\u00B5s] = ", (end - start) * 100000)
#assert(mean_arithmetic == 3.8)
print ("==================================================================================")

print ("==================================================================================")
# Arrange
start = timeit.default_timer()
# Act
median_high = statistics.median_high(data)
end = timeit.default_timer()
# Assert
print ("median_high                   = ", median_high)
print (u"              time [\u00B5s] = ", (end - start) * 100000)
#assert(mean_arithmetic == 3.8)
print ("==================================================================================")

print ("==================================================================================")
# Arrange
start = timeit.default_timer()
# Act
median_grouped = statistics.median_grouped(data)
end = timeit.default_timer()
# Assert
print ("median_grouped                = ", median_grouped)
print (u"              time [\u00B5s] = ", (end - start) * 100000)
#assert(mean_arithmetic == 3.8)
print ("==================================================================================")

